integrations/airflow/marquez_airflow/dag.py

```diff
@@ -125,60 +126,60 @@ def handle_callback(self, *args, **kwargs):
 
         return super().handle_callback(*args)
 
-    def _report_task_instances(self, dagrun, run_args, session):
+    def _report_task_instances(self, dagrun, session):
         task_instances = dagrun.get_task_instances()
-        for ti in task_instances:
+        for task_instance in task_instances:
             try:
-                self._report_task_instance(ti, dagrun, run_args, session)
+                self._report_task_instance(task_instance, dagrun, session)
             except Exception as e:
                 self.log.error(
                     f'Failed to record task instance: {e} '
                     f'dag_id={self.dag_id}',
                     exc_info=True)
 
-    def _report_task_instance(self, ti, dagrun, run_args, session):
-        task = self.get_task(ti.task_id)
-        run_ids = JobIdMapping.pop(
-            self._marquez_job_name_from_ti(ti), dagrun.run_id, session)
-        steps = self._extract_metadata(dagrun, task, ti)
+    def _report_task_instance(self, task_instance, dagrun, session):
+        task = self.get_task(task_instance.task_id)
+        task_run_id = JobIdMapping.pop(
+            self._marquez_job_name_from_task_instance(task_instance), dagrun.run_id, session)
+        step = self._extract_metadata(dagrun, task, task_instance)
 
         # Note: run_ids could be missing if it was removed from airflow
         # or the job could not be registered.
-        if not run_ids:
-            [_MARQUEZ.create_job(
-                step, self._get_location(task), self.description)
-             for step in steps]
-            run_ids = [_MARQUEZ.create_run(
-                self.new_run_id(),
+        job_name = self._marquez_job_name(self.dag_id, task.task_id)
+
+        if not task_run_id:
+            task_run_id = _MARQUEZ.start_task(
+                job_name,
+                self.description,
+                DagUtils.to_iso_8601(self._now_ms()),
+                None,  # TODO: add parent hierarchy
+                self._get_location(task),
+                DagUtils.to_iso_8601(task_instance.start_date),
+                DagUtils.to_iso_8601(task_instance.end_date),
                 step,
-                run_args,
-                DagUtils.to_iso_8601(ti.start_date),
-                DagUtils.to_iso_8601(ti.end_date)
-            ) for step in steps]
-            if not run_ids:
+            )
```

Comment:
this looks duplicated (and slightly different) from line 90

Reply:
The original assumption that I kept is that 

    # Note: task_run_id could be missing if it was removed from airflow
    # or the job could not be registered.

Then we emit START event again, as we OpenLineage contract specifies that we need to have 1 start event.
If the assumption is not true, then it could be removed I think.